core/transaction.c

```diff
@@ -471,3 +481,27 @@ void transaction_step(lwm2m_context_t * contextP,
         transacP = nextP;
     }
 }
+
+void transaction_set_payload(lwm2m_transaction_t * transaction, uint8_t * buffer, int length)
+{
+    transaction->payload = buffer;
+    transaction->payload_len = length;
+
+    if ( length > REST_MAX_CHUNK_SIZE ){
+        coap_set_header_block1(transaction->message, 0, true, REST_MAX_CHUNK_SIZE);
+    }
+
+    coap_set_payload(transaction->message, buffer, MIN(length, REST_MAX_CHUNK_SIZE));
+}
+
+bool transaction_free_userData(lwm2m_context_t * context, lwm2m_transaction_t * transaction)
+{
+    lwm2m_transaction_t * target = context->transactionList;
```

Comment:
Check context == NULL

Reply:
I think we had similar discussions in other cases and the argument beeing that in order to keep a minimal footprint we need to trust calling functions and callback to not do anything vile.